src/domain_classes/tree_node.py

```diff
@@ -266,12 +266,21 @@ def __init__(
         )
         self.entity: dict = entity if entity else {}
         self.error_message = None
+        if (
+            entity is not None
+            and entity != {}
+            and entity["type"] == "dmss://system/SIMOS/Link"
+            and "targetName" in entity
+        ):
+            self.type = entity["targetType"]
```

Comment:
Are we sure we want to "fake" which type this node is?
Seems like that could lead to some issues.
What kind of problems do we have that deems this necessary?

Also;
- Should use an enum for the "type" check,
- And "if entity" instead of "is not none" and "!= {}"